lib/matplotlib/axes/_axes.py

```diff
@@ -6054,25 +6054,30 @@ def _normalize_input(inp, ename='input'):
             """
             if (isinstance(x, np.ndarray) or
                     not iterable(cbook.safe_first_element(inp))):
-                # TODO: support masked arrays;
-                inp = np.asarray(inp)
+
+                inp = cbook.safe_masked_invalid(inp)
                 if inp.ndim == 2:
                     # 2-D input with columns as datasets; switch to rows
                     inp = inp.T
+
+                    if inp.shape[1] < inp.shape[0]:
```

Comment:
I now see that this is just moving this warning around.  I would be in favor of removing it, but if it is already in the code base, mostly ignore my previous comment.